src/types.rs

```diff
@@ -97,6 +97,59 @@ pub trait IntoTableRef {
     fn into_table_ref(self) -> TableRef;
 }
 
+#[derive(Clone, Debug)]
+pub struct TypeRef {
```

Comment:
I though about this and have two idea:
1. This should be part of `extension/postgres` not here
2. Maybe  is better rewrite it to enum:
```rust
pub enum TypeRef {
    Type(...),
    DatabaseType(...),
    DatabaseSchemaType(...)
}
```

Reply:
Sounds good. I'll move it to `extension/postgres`!